src/consensus/versionbits.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VERSIONBITS_H
+#define BITCOIN_VERSIONBITS_H
+
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <stdint.h>
+
+namespace Consensus
+{
+
+const int           NO_RULE      = -1;
+
+namespace VersionBits
+{
+
+const int           VERSION_HIGH_BITS   = 0x20000000;
+const int           VERSION_BITS_MASK   = 0x1fffffff;
+const char          MIN_BIT             = 0;
+const char          MAX_BIT             = 28;
+
+enum RuleState { UNDEFINED, DEFINED, LOCKED_IN, ACTIVE, FAILED };
+
+const char* GetRuleStateText(int ruleState, bool bUseCaps = false);
+
+bool UsesVersionBits(int nVersion);
+
+class SoftFork
+{
+public:
+    SoftFork(int bit, int rule, int threshold, uint32_t deployTime, uint32_t expireTime) :
+        m_bit(bit),
+        m_rule(rule),
+        m_threshold(threshold),
+        m_deployTime(deployTime),
+        m_expireTime(expireTime) { }
+
+    int         GetBit() const { return m_bit; }
+    int         GetRule() const { return m_rule; }
+    int         GetThreshold() const { return m_threshold; }
+    uint32_t    GetDeployTime() const { return m_deployTime; }
+    uint32_t    GetExpireTime() const { return m_expireTime; }
+
+private:
+    int m_bit;
+    int m_rule;
+    int m_threshold;
```

Comment:
Maybe it would be useful to plan for 10 bits 95%, 10 bits 85%, 5 bits 75%, and 5 bits 65%? Does anyone expect to have more than 10 ongoing softforks at a time anyway?